ldap/servers/plugins/acctpolicy/acct_plugin.c

```diff
@@ -130,9 +133,97 @@ acct_inact_limit(Slapi_PBlock *pb, const char *dn, Slapi_Entry *target_entry, ac
     return (rc);
 }
 
+/*
+  Preserve bind time stamps in virtual attribute
+*/
+int
+acct_update_login_history(const char *dn, char *timestr)
+{
+    void *plugin_id = NULL;
+    int rc = -1;
+    int num_entrys = 0;
+    size_t i = 0;
+    char **login_hist = NULL;
+    Slapi_PBlock *entry_pb = NULL;
+    Slapi_PBlock *mod_pb;
+    Slapi_Entry *e = NULL;
+    Slapi_DN *sdn = NULL;
+    acctPluginCfg *cfg;
+    LDAPMod attribute;
+    LDAPMod *list_of_mods[2];
+
+    plugin_id = get_identity();
+
+    sdn = slapi_sdn_new_normdn_byref(dn);
+    slapi_search_get_entry(&entry_pb, sdn, NULL, &e, plugin_id);
+
+    config_rd_lock();
+    cfg = get_config();
+
+    if (!timestr) {
+        return (rc);
+    }
```

Comment:
I worry that we may not unlock the config if we return here. Or is it resolved later?